definitions/features/downstream.rb

```diff
@@ -55,24 +55,33 @@ def rhsm_refresh
   def rh_repos(sat_version)
     sat_version = version(sat_version)
     rh_version_major = execute!('facter operatingsystemmajrelease')
-    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
+    rh_repos = main_rh_repos(rh_version_major)
 
+    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
     sat_repo_id = "rhel-#{rh_version_major}-server-satellite-#{sat_version_full}-rpms"
     sat_tools_repo_id = "rhel-#{rh_version_major}-server-satellite-tools-#{sat_version_full}-rpms"
-    # Override to use Beta repositories for 6.4 until GA
+
     if sat_version.to_s == '6.4'
+      # Override to use Beta repositories for 6.4 until GA
       sat_repo_id = "rhel-server-#{rh_version_major}-satellite-6-beta-rpms"
-    end
 
-    rh_repos = ["rhel-#{rh_version_major}-server-rpms",
-                "rhel-server-rhscl-#{rh_version_major}-rpms",
-                "rhel-#{rh_version_major}-server-satellite-maintenance-6-rpms",
-                sat_tools_repo_id,
-                sat_repo_id]
-    rh_repos << 'rhel-7-server-ansible-2-rpms' if sat_version.to_s == '6.4'
+      rh_repos << 'rhel-7-server-ansible-2-rpms'
+    end
+    rh_repos.concat([sat_tools_repo_id, sat_repo_id])
+    rh_repos << 'rhel-7-server-satellite-tools-6.3-puppet4-rpms' if puppet_repo?(sat_version)
     rh_repos
   end
 
+  def main_rh_repos(rh_version_major)
+    ["rhel-#{rh_version_major}-server-rpms",
+     "rhel-server-rhscl-#{rh_version_major}-rpms",
+     "rhel-#{rh_version_major}-server-satellite-maintenance-6-rpms"]
+  end
+
+  def puppet_repo?(sat_version)
+    sat_version.to_s >= '6.3' && (execute('rpm -q puppet') =~ /puppet-4/)
```

Comment:
Please use `package_version` helper instead of direct `execute` https://github.com/theforeman/foreman_maintain/blob/1ae71c3ad827503eb370310a6b27232f80289351/lib/foreman_maintain/concerns/system_helpers.rb#L88